docs/design/storage-auto-scale.md

```diff
@@ -0,0 +1,83 @@
+# Storage Automatic Scaler
+
+## Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices Osds so this feature helps in automatic scaling. 
+
+## Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+## High-Level Design
+
+- Create a StorageAutoScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    kind: StorageAutoScaling
+    metadata:
+        name: <storageclusterName>-<deviceClassName>
+        namespace: openshift-storage
+    spec:
+        storageCluster: 
+            name: <storageclusterName>
+        deviceClassName: ssd
+        clusterCapacityLimit: <capacity>
+        maxOsdSize: 8Ti
+        timeoutSeconds: 1800   
+    status:
+        phase: (NotStarted | InProgress | Failed | Succeeded)
+        LastRunTimestamp: (only on Failed | Succeeded)
```

Comment:
This timestamp is confusing. If it's failed, that just means the most recent expansion hasn't succeeded yet. It could still succeed if there is some issue resolved in the cluster (add more nodes, add more mem/cpu, etc). What about splitting to these two times? As a naming convention, I also recommend "Time" instead of "Timestamp"
```yaml
    lastExpansionStartTime: (always set after a new expansion is started)
    lastExpansionCompletionTime: (only set on success)